plugin/src/init.server.lua

```diff
@@ -1,222 +1,289 @@
-	SettingsModule.Parent = AnalyticsService
-end
-assert(SettingsModule, "failed to create settings module")
-
-local function LoadSettings()
-	local result, parseError: any = loadstring(SettingsModule.Source)
-	if result == nil then
-		warn("[Luau Language Server] Could not load settings: " .. parseError)
-		Settings = nil
-		return
-	end
-	assert(result, "")
-	local _, err = pcall(function()
-		Settings = result()
-	end)
-	if err then
-		warn(err)
-	end
-	if type(Settings) ~= "table" then
-		Settings = nil
-		warn("[Luau Language Server] Could not load settings: invalid settings")
-	elseif type(Settings.port) ~= "number" then
-		Settings = nil
-		warn("[Luau Language Server] Could not load settings: invalid port")
-	elseif type(Settings.startAutomatically) ~= "boolean" then
-		Settings = nil
-		warn("[Luau Language Server] Could not load settings: invalid startAutomatically value")
-	elseif type(Settings.include) ~= "table" then
-		Settings = nil
-		warn("[Luau Language Server] Could not load settings: invalid include list")
-	end
-end
-
-LoadSettings()
-
---#endregion
+local SETTINGS_MODULE_NAME = "LuauLanguageServer_Settings"
+
+local DEFAULT_SOURCE = [[
+return {
+    --// Should be unique to Rojo port (different).
+	Port = 3667,
+
+    --// Decides whether or not the companion plugin automatically starts listening on studio launch.
+	StartAutomatically = false,
+
+    --// Setting to true will enable verbose error messages and warns. Currently does nothing.
+    DebugMode = false,
+
+    --// List of instances who's descendants will be encoded and sent to VS Code.
+	Include = {
+		game:GetService("Workspace"),
+		game:GetService("Players"),
+		game:GetService("Lighting"),
+		game:GetService("ReplicatedFirst"),
+		game:GetService("ReplicatedStorage"),
+		game:GetService("ServerScriptService"),
+		game:GetService("ServerStorage"),
+		game:GetService("StarterGui"),
+		game:GetService("StarterPack"),
+		game:GetService("StarterPlayer"),
+		game:GetService("SoundService"),
+		game:GetService("Chat"),
+		game:GetService("LocalizationService"),
+		game:GetService("TestService"),
+	},
+}]]
+
+local Connections: { RBXScriptConnection } = {}
+local Connected = false
 
---#region Connect
+type EncodedInstance = {
+	Name: string,
+	ClassName: string,
+	Children: { EncodedInstance? },
+}
 
+local Toolbar = plugin:CreateToolbar("Luau Language Server")
 local ConnectAction = plugin:CreatePluginAction(
 	"Luau Language Server Connect",
 	"Connect",
 	"Connects to Luau Language Server",
 	"rbxassetid://11115506617",
 	true
 )
+local ConnectButton = Toolbar:CreateButton(
+	"Luau Language Server Setup",
+	"Toggle Menu",
+	"rbxassetid://11115506617",
+	"Luau Language Server"
+)
+local SettingsButton = Toolbar:CreateButton("Settings", "Open Settings", "rbxassetid://13997395868")
 
-local connected = Instance.new("BoolValue")
-local connections = {}
+local DefaultSettingsModule = Instance.new("ModuleScript")
+DefaultSettingsModule.Name = SETTINGS_MODULE_NAME
```

Comment:
What is the point in inlining this script contents and then creating an instance. Was something wrong with the predefined `DefaultSettings.lua`? This seems like a roundabout way to end up just doing the same thing that was there before.